src/Collapse/include/gudhi/Flag_complex_sparse_matrix.h

```diff
@@ -0,0 +1,424 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Siddharth Pritam
+ *
+ *    Copyright (C) 2018 Inria
+ *
+ *    Modification(s):
+ *      - 2020/03 Vincent Rouvreau: integration to the gudhi library
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+#ifndef FLAG_COMPLEX_SPARSE_MATRIX_H_
+#define FLAG_COMPLEX_SPARSE_MATRIX_H_
+
+#include <gudhi/graph_simplicial_complex.h>
+
+#include <boost/functional/hash.hpp>
+#include <boost/graph/adjacency_list.hpp>
+
+#include <Eigen/Sparse>
+
+#ifdef GUDHI_USE_TBB
+#include <tbb/parallel_sort.h>
+#endif
+
+#include <iostream>
+#include <utility>  // for std::pair
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+#include <set>
+#include <tuple>  // for std::tie
+#include <algorithm>  // for std::includes
+#include <iterator>  // for std::inserter
+
+namespace Gudhi {
+
+namespace collapse {
+
+/**
+ * \class Flag_complex_sparse_matrix
+ * \brief Flag complex sparse matrix data structure.
+ *
+ * \ingroup collapse
+ *
+ * \details
+ * A class to store the vertices v/s MaxSimplices Sparse Matrix and to perform collapse operations using the N^2()
+ * Algorithm.
+ *
+ * \tparam Vertex type must be a signed integer type. It admits a total order <.
+ * \tparam Filtration type for the value of the filtration function. Must be comparable with <.
+ */
+template<typename Vertex, typename Filtration>
+class Flag_complex_sparse_matrix {
+ public:
+  /** \brief Re-define Vertex as Vertex_handle type to ease the interface with compute_proximity_graph. */
+  using Vertex_handle = Vertex;
+  /** \brief Re-define Filtration as Filtration_value type to ease the interface with compute_proximity_graph. */
+  using Filtration_value = Filtration;
+  /** \brief This is an ordered pair, An edge is stored with convention of the first element being the smaller i.e
+   * {2,3} not {3,2}. However this is at the level of row indices on actual vertex lables.
+   */
+  using Edge = std::pair<Vertex_handle, Vertex_handle>;
+
+ private:
+  // Row_index type in the sparse matrix
+  using Row_index = std::size_t;
+
+  // The sparse matrix data type
+  using Sparse_row_matrix = Eigen::SparseMatrix<Filtration_value, Eigen::RowMajor>;
+
+  // A range of row indices
+  using Row_indices_vector = std::vector<Row_index>;
+
+ public:
+  /** \brief A Filtered_edge is a std::pair<std::pair<`Vertex_handle`, `Vertex_handle`>, `Filtration_value`>. */
+  using Filtered_edge = std::pair<Edge, Filtration_value>;
+  /** \brief Proximity_graph is a type that can be used to construct easily a Flag_complex_sparse_matrix. */
+  using Proximity_graph = Gudhi::Proximity_graph<Flag_complex_sparse_matrix>;
```

Comment:
I can't find Gudhi::Proximity_graph in the doc.

Reply:
compute_proximity_graph is now available in the documentation and search. If you want a link to it, I have to set `EXTRACT_ALL = YES`in `Doxyfile.in`. The side effect of it is that it generates documentation even for `EXCLUDE` list (hera, python, ...)